app/src/pages/Lobby/index.tsx

```diff
@@ -209,13 +211,44 @@ function WaitingRoomSection() {
 
 function Lobby() {
   const currentPlayer = React.useContext(CurrentPlayerContext)
+  const [updatePlayer] = useUpdatePlayerMutation()
   const classes = useStyles()
+  const titleClasses = useTitleStyle()
+  const location = useLocation()
+  const [hideShare, setHideShare] = React.useState(false)
+
+  React.useEffect(() => {
+    const params = new URLSearchParams(location.search)
+
+    setHideShare(params.get("hideshare") === "true")
+
+    const username = params.get("name")
+    if (username) {
+      updatePlayer({
+        variables: {
+          id: currentPlayer.id,
+          input: { username },
+        },
+      })
+    }
+  }, [])
 
   return (
     <div>
-      <div className={classes.section}>
-        <ShareSection />
-      </div>
+      {!hideShare ? (
+        <div className={classes.section}>
+          <ShareSection />
+        </div>
```

Comment:
I assume there'll be another place on Rocket Crab that displays the code to share?

Reply:
Yes. But, rocketcrab uses it's own code and the fishbowl code will only be used internally. Here's how it works for Drawphone. On the left, using Drawphone without rocketcrab, drawphone shows its code to the user. On the right, using rocketcrab, only the rocketcrab code is shown and Drawphone's code is hidden so as not to confuse the user. Sorry if that's a little confusing 😕 

<img src="https://user-images.githubusercontent.com/8444595/97715628-a1f91300-1a90-11eb-9661-a01b575ef0a4.png" width="200" /><img src="https://user-images.githubusercontent.com/8444595/97715633-a291a980-1a90-11eb-86ae-f982d255ea21.png" width="200" />

Reply:
Make sense! 🔥